specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json

```diff
@@ -6336,17 +6340,47 @@
       "properties": {
         "id": {
           "type": "string",
-          "description": "The security posture reference id in the form of /CommunityGalleries/{communityGalleryName}/securityPostures/{securityPostureName}/versions/{major.minor.patch}|{major.*}|latest"
+          "description": "The security posture reference id in the form of /CommunityGalleries/{communityGalleryName}/securityPostures/{securityPostureName}/versions/{major.minor.patch}|latest"
         },
         "excludeExtensions": {
           "type": "array",
           "items": {
-            "$ref": "./virtualMachine.json#/definitions/VirtualMachineExtension"
+            "type": "string"
           },
-          "description": "List of virtual machine extensions to exclude when applying the Security Posture."
+          "description": "The list of virtual machine extension names to exclude when applying the security posture."
+        },
+        "extensionsSettings": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "The name of the virtual machine extension included in the security posture."
+              },
+              "settings": {
+                "type": "object",
```

Comment:
>object [](http://example.com/codeflow?start=25&length=6)

settings and protectedSettings need to be more defined than type:object. details - [https://github.com/Azure/azure-openapi-validator/blob/bf62821cb237f949470fa9f75a7db86d9e218d4c/docs/properties-type-object-no-definition.md](https://github.com/Azure/azure-openapi-validator/blob/bf62821cb237f949470fa9f75a7db86d9e218d4c/docs/properties-type-object-no-definition.md)

Reply:
These properties are a mirror from what's available under the `VirtualMachineScaleSetExtensionsProperties` definition in the same schema. (ref. [Line 5002](https://github.com/Azure/azure-rest-api-specs/blob/8aa8b9c4012351e5d47d09ca8bd80819804bcba0/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json#L5002)). I believe these are dynamic depending on what the VM Extension is - how do you recommend unblocking this?

Reply:
So if there is no schema for this object, how does the RP or the client use it? How do they know how to analyze this object? What is your RP going to actually do with this property? As of now, it's undocumented, unversioned, and subject to change at any time. This is a very brittle design.

Reply:
We've decided to move forward without this property for now. For next version, we'll work with the VM Extension team to understand why there's not a defined schema for `settings` and `protectedSettings` but I believe it's because they're dynamic depending on the VM Extension and are ingested all as optional within CRP code and handled as such.